src/components/transport_manager/include/transport_manager/cloud/websocket_client_connection.h

```diff
@@ -126,14 +139,16 @@ class WebsocketClientConnection
  private:
   TransportAdapterController* controller_;
   boost::asio::io_context ioc_;
+  ssl::context ctx_;
   tcp::resolver resolver_;
-  websocket::stream<tcp::socket> ws_;
   boost::beast::flat_buffer buffer_;
   std::string host_;
   std::string text_;
+  boost::variant<WSptr, WSSptr> dynamic_ws_;
```

Comment:
I guess the only benefit of the `boost::variant` in this situation is that it can store either the `websocket::stream<tcp::socket>` or the `websocket::stream<ssl::stream<tcp::socket>>`. This way I only have to create one shared pointer for either type. 

I agree that having to cast it every time makes it use the same amount of code as switching between two different objects

Reply:
I believe that it would honestly be cleaner having two variables, since it would remove the casting code. I don't believe that this overhead is worth saving one pointer worth of space.